bundles/org.openhab.binding.fronius/src/main/java/org/openhab/binding/fronius/internal/action/FroniusSymoInverterActions.java

```diff
@@ -107,56 +108,62 @@ public void setThingHandler(@Nullable ThingHandler handler) {
     }
 
     @RuleAction(label = "@text/actions.reset-battery-control.label", description = "@text/actions.reset-battery-control.description")
-    public void resetBatteryControl() {
+    public @ActionOutput(type = "boolean", label = "Success") boolean resetBatteryControl() {
         FroniusSymoInverterHandler handler = this.handler;
         if (handler != null) {
-            handler.resetBatteryControl();
+            return handler.resetBatteryControl();
         }
+        return false;
     }
 
     @RuleAction(label = "@text/actions.hold-battery-charge.label", description = "@text/actions.hold-battery-charge.description")
-    public void holdBatteryCharge() {
+    public @ActionOutput(type = "boolean", label = "Success") boolean holdBatteryCharge() {
         FroniusSymoInverterHandler handler = this.handler;
         if (handler != null) {
-            handler.holdBatteryCharge();
+            return handler.holdBatteryCharge();
         }
+        return false;
     }
 
     @RuleAction(label = "@text/actions.add-hold-battery-charge-schedule.label", description = "@text/actions.add-hold-battery-charge-schedule.description")
-    public void addHoldBatteryChargeSchedule(
-            @ActionInput(name = "from", label = "@text/actions.from.label", description = "@text/actions.from.description") LocalTime from,
-            @ActionInput(name = "until", label = "@text/actions.until.label", description = "@text/actions.until.description") LocalTime until) {
+    public @ActionOutput(type = "boolean", label = "Success") boolean addHoldBatteryChargeSchedule(
+            @ActionInput(name = "from", label = "@text/actions.from.label", description = "@text/actions.from.description", required = true) LocalTime from,
+            @ActionInput(name = "until", label = "@text/actions.until.label", description = "@text/actions.until.description", required = true) LocalTime until) {
         FroniusSymoInverterHandler handler = this.handler;
         if (handler != null) {
-            handler.addHoldBatteryChargeSchedule(from, until);
+            return handler.addHoldBatteryChargeSchedule(from, until);
         }
+        return false;
     }
 
-    public void addHoldBatteryChargeSchedule(ZonedDateTime from, ZonedDateTime until) {
-        addHoldBatteryChargeSchedule(from.toLocalTime(), until.toLocalTime());
+    public boolean addHoldBatteryChargeSchedule(ZonedDateTime from, ZonedDateTime until) {
```

Comment:
Annotations are missing for this action.

Reply:
This is unusual but ok.

Reply:
Not sure which binding it was but not annotation any existing override helps improving the UI experience.